.github/workflows/ci.yml

```diff
@@ -48,3 +55,31 @@ jobs:
       - name: Check TinyGo version number consistency
         run: |
           builder/docker/tinygo/sanity.sh
+
+  # only run if reference is a tag
+  release:
+    if: startsWith(github.ref, 'refs/tags/v')
+
+    needs: [lint, test]
+    runs-on: ubuntu-latest
```

Comment:
Will only run this step if the target/reference is a tag (which implies release or pre-release). This step will also only run if both `lint` and `test` steps have passed.